.github/workflows/shellcheck.yml

```diff
@@ -20,6 +20,7 @@ jobs:
         uses: actions/checkout@v3
```

Comment:
perhaps we can bump this to v4 as part of this?

Reply:
I know very little about Gihub Actions - `persist-credentials` felt simple and understandable to me, but I have much worse idea about changes related to bumping actions to v4. I don't mind, but someone else need to decide.

Reply:
Aha! Now I get it: it's the last `actions/checkout` using `v3`. Fixing in e61ff666.